schema/loader.go

```diff
@@ -0,0 +1,144 @@
+package schema
+
+import (
+	"embed"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/santhosh-tekuri/jsonschema/v5"
+)
+
+func init() {
+	// load all local schemas
+	if err := LoadAllLocal(); err != nil {
+		panic(err)
+	}
+}
+
+var (
+	//go:embed known_schemas
+	knownSchemas embed.FS
+)
+
+type (
+	SchemaFile string
+)
+
+const (
+	schemaDirectory = "known_schemas/"
+
+	// Presentation Exchange Schemas
+	PresentationDefinitionSchema              SchemaFile = "pe-presentation-definition.json"
+	PresentationDefinitionEnvelopeSchema      SchemaFile = "pe-presentation-definition-envelope.json"
+	PresentationSubmissionSchema              SchemaFile = "pe-presentation-submission.json"
+	PresentationClaimFormatDesignationsSchema SchemaFile = "pe-definition-claim-format-designations.json"
+	SubmissionClaimFormatDesignationsSchema   SchemaFile = "pe-submission-claim-format-designations.json"
+	SubmissionRequirementSchema               SchemaFile = "pe-submission-requirement.json"
+	SubmissionRequirementsSchema              SchemaFile = "pe-submission-requirements.json"
+
+	PresentationClaimFormatDesignationFormatDefinition SchemaFile = "pe-submission-claim-format-designations.json"
+
+	// Credential Manifest Schemas
+	CredentialManifestSchema    SchemaFile = "cm-credential-manifest.json"
+	CredentialApplicationSchema SchemaFile = "cm-credential-application.json"
+	CredentialResponseSchema    SchemaFile = "cm-credential-response.json"
+	OutputDescriptorsSchema     SchemaFile = "cm-output-descriptors.json"
+
+	// Wallet Rendering Schemas
+	DisplayMappingObjectSchema        SchemaFile = "wr-display-mapping-object.json"
+	EntityStylesSchema                SchemaFile = "wr-entity-styles.json"
+	LabeledDisplayMappingObjectSchema SchemaFile = "wr-labeled-display-mapping-object.json"
+
+	// VC JSON Schema Schemas
+	VerifiableCredentialJSONSchemaSchema SchemaFile = "vc-json-schema.json"
+)
+
+func (s SchemaFile) String() string {
+	return string(s)
+}
+
+// LocalLoader is a struct that holds local schemas
+type LocalLoader struct {
+	localSchemas map[string]string
+}
+
+// NewLocalLoader returns a new LocalLoader without any schemas to be loaded locally
+// Calling this method disables the ability to load schemas remotely over http and https
+func NewLocalLoader() LocalLoader {
+	localSchemas := make(map[string]string)
+	jsonschema.Loaders["https"] = func(url string) (io.ReadCloser, error) {
+		schema, ok := localSchemas[strings.TrimPrefix(url, "https://")]
+		if !ok {
+			return nil, fmt.Errorf("%q not found", url)
+		}
+		return io.NopCloser(strings.NewReader(schema)), nil
+	}
+	jsonschema.Loaders["http"] = func(url string) (io.ReadCloser, error) {
+		schema, ok := localSchemas[strings.TrimPrefix(url, "http://")]
+		if !ok {
+			return nil, fmt.Errorf("%q not found", url)
+		}
+		return io.NopCloser(strings.NewReader(schema)), nil
+	}
```

Comment:
I have a couple of suggestions for your consideration:
* Rework `LocalLoader` into `LocalLoaderFactory`.
* Remove the `jsonschema.Loader` directives from this file, and the `init` functions (let users of the SDK do it themselves).
* Provide a method for creating loaders, something like what's below. 

```go
type Loader func(url string) (io.ReadCloser, error)

func (l *LocalLoaderFactory) ProtocolLoader(protocol string) Loader {
  return func(url string) (io.ReadCloser, error) {
		schema, ok := localSchemas[strings.TrimPrefix(url, protocol + "://")]
		if !ok {
			return nil, fmt.Errorf("%q not found", url)
		}
		return io.NopCloser(strings.NewReader(schema)), nil
	}
}
```

Once that is done, then user's of the SDK can decide where to override the `jsonschema` loader. 

```go
// Somewhere in the ssi-service

loaderFactory := schema.NewLocalLoaderFactory()
jsonschema.Loader["http"] = loaderFactory.ProtocolLoader("http")
jsonschema.Loader["https"] = loaderFactory.ProtocolLoader("https")
```

Thoughts?

Reply:
How about defining the TestMain (as described in this https://pkg.go.dev/testing#hdr-Main)? Within that function we can patch the jsonschema lib.